File: CarePackage/Software/EaApp.cs
namespace CarePackage.Software;

public class EaApp : BaseSoftware
{
    public override string           Key            { get; set; } = "ea-app";
    public override string           Name           { get; set; } = "EA App";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Gaming;
    public override string           DownloadName   { get; set; } = "EAappInstaller.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = false;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.ea_app;
    public override string           Homepage       { get; set; } = "https://ea.com/ea-app";

    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://origin-a.akamaihd.net/EA-Desktop-Client-Download/installer-releases/EAappInstaller.exe");
}

File: packages/app/src/services/software/definitions/eaApp.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'ea-app';
	public name = 'EA App';
	public category = [SoftwareCategory.Gaming];
	public downloadName = 'EAappInstaller.exe';
	public icon = 'ea-app.png';
	public homepage = 'https://ea.com/ea-app';

	public async resolveDownloadUrl() {
		return ok('https://origin-a.akamaihd.net/EA-Desktop-Client-Download/installer-releases/EAappInstaller.exe');
	}
}
